src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -2964,6 +2826,376 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+  // CTR AES crypt.
+  // Arguments:
+  //
+  // Inputs:
+  //   c_rarg0   - source byte array address
+  //   c_rarg1   - destination byte array address
+  //   c_rarg2   - K (key) in little endian int array
+  //   c_rarg3   - counter vector byte array address
+  //   c_rarg4   - input length
+  //   c_rarg5   - saved encryptedCounter start
+  //   c_rarg6   - saved used length
+  //
+  // Output:
+  //   r0       - input length
+  //
+  address generate_counterMode_AESCrypt() {
+    const Register in = c_rarg0;
+    const Register out = c_rarg1;
+    const Register key = c_rarg2;
+    const Register counter = c_rarg3;
+    const Register saved_len = c_rarg4, len = r10;
+    const Register saved_encrypted_ctr = c_rarg5;
+    const Register used_ptr = c_rarg6, used = r12;
+
+    const Register offset = r7;
+    const Register keylen = r11;
+
+    const unsigned char block_size = 16;
+    const int bulk_width = 4;
+
+
+    // Algorithm:
+    //
+    // int result = len;
+    // while (len-- > 0) {
```

Comment:
I see that algorithm code comes from CounterMode.implCrypt, but  while (len-- > 0) seems not to be exactly same as algorithm here. I think it should be `while (len > 0)`

`blockSize()` at line 2865 should be `blockSize`

Reply:
Done.